EShop.Web/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,43 @@
+﻿using EShop.Web.Models;
+using EShop.Web.Services;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace EShop.Web.Controllers
+{
+    public class ProductsController : Controller
+    {
+        public ProductsController(IRepository<ProductModel> product)
+        {
+            this.ProductRepository = product;
+        }
+        public IRepository<ProductModel> ProductRepository;
+        
+        public IActionResult Index()
+        {
+            var produse = ProductRepository.GetAll();
+            return View(produse);
```

Comment:
`produse` nu e folosit in mai mult de o locatie, este Ok sa dai return `View(ProductRepository.GetAll());`

De asemenea as vrea de acum inainte cand accesezi ceva de pe `this`, sa scri `this.` inainte.

In cazul de fata  `this.View`, `this.ProductRepository`.